lib/param_parsing/match_version_requirement.go

```diff
@@ -0,0 +1,42 @@
+//nolint:revive // FIXME: don't use an underscore in package name
+package param_parsing
+
+import "github.com/warrensbox/terraform-switcher/lib"
+
+// MatchVersionRequirement : Check if a given version meets a version requirement mandated by the configuration.
+// - Return `0` if the version matches the requirement, otherwise return `2`.
+// - Return `1` for any kind of general error (like version format parsing error).
+func MatchVersionRequirement(parameters Params) int {
```

Comment:
It's probably more normal to return a bool (matches) and an error with the error.
I'm not returning numbers here makes sense (as I don't see more than 2 options, other than error).
But, if you do stick to returning numbers, you definitely need to return some sort of Enum - numbers (or strings) are just too magic and don't provide a logical way to validate calling behaoir (other than reading the comment)

Reply:
Given that both of you and @warrensbox made a notice on this behavior, let me work on improving this bit. Thanks.